File: acmp.ru/[0701] - [0800]/[0792] Торговые сделки в Амбере.cpp
#include <iostream>
using namespace std;

int count(int number, int base) {
    int cnt = 0;

    while (number != 0) {
        cnt += number % base;
        number /= base;
    }

    return cnt;
}

int main() {
    int n1, p1, k1, n2, p2, k2;
    cin >> n1 >> p1 >> n2 >> p2;

    k1 = count(n1, p1);
    k2 = count(n2, p2);

    if (k1 == k2)
        cout << k1 << endl;
    else
        cout << 0 << endl;

    return 0;
}

File: acmp.ru/[0701] - [0800]/[0792] Торговые сделки в Амбере.py
def count(number, base):
    cnt = 0

    while number != 0:
        cnt += number % base
        number //= base

    return cnt


k1 = count(*map(int, input().split()))
k2 = count(*map(int, input().split()))

print(k1 if k1 == k2 else 0)
